### ISSUE TITLE
True xDS version in admin interface

### ISSUE THREAD

**Issue by htuch** [envoyproxy/envoy#2172]
While we added config hash for versioning of xDS in https://github.com/envoyproxy/envoy/pull/1805, it would be useful for integration testing to have the actual version info as understood by the management server and xDS exposed on an admin interface handler. This won't be available via stats, since version info is an arbitrary length string which we can't represent as a gauge, so just an admin handler.

**Comment by htuch** [comment]
@rfaulk @mattklein123 I'm thinking of adding for ADS a JSON/proto handler for this that returns some message defined in data-plane-api describing each of the muxed protocols. For non-ADS, it's a bit more complicated for RDS/EDS, since each of these xDS might actually have multiple versions, as resources may fan out to multiple servers individually. This is in aid of Google integration testing, I think others will find it helpful.

LMK if the above proposal makes sense and I can implement.

**Comment by mattklein123** [comment]
SGTM. Should this be the first admin API that we define a proto response for instead of hand rolling JSON output? Incrementally more work but would be cool.

**Comment by junr03** [comment]
For posterity: right now we have the version info string of different xDS APIs surfaced in a couple places:
1. RDS: in the `/routes` admin endpoint, for each route configuration. You are right that it would be a little bit more complicated to do so centrally for RDS. Which is why I did the handler work in the scope of the router.
2. CDS: in the `/clusters` admin endpoint, for the cluster manager.

@htuch what do you think about removing these from their separate places and unifying them under the one `/xds`(?) admin endpoint? I would love to see the admin endpoint output converted into proto messages, as I have already been burned by bad JSON in the /routes output.

**Comment by htuch** [comment]
@junr03 Yep, I think moving to a single handler would be great. Let's make this happen, hopefully I get cycles this year unless someone is interested in this before Q1.
